Reduce javadoc errors by adding missing dependencies

pleku opened the pull request:
vaadin-all currently has +100 errors/warning when creating javadocs,
this reduces most of them related to missing resources to link to.
Also updated liferay dependency to 6.2.5 from 6.0.2 for vaadin-server,
since FW 8 supports liferay 6.2.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/7959)
<!-- Reviewable:end -->

hesara commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pom.xml, line 26 at r1](https://reviewable.io:443/reviews/vaadin/framework/7959#-KYnF3xWGOW25bZFBWYU:-KYnF3xWGOW25bZFBWYV:b1h21jj) ([raw file](https://github.com/vaadin/framework/blob/97dcf58219a3ddfe7d9404582f42865d1deabc10/pom.xml#L26)):*
> ```XML
> 
>         <!-- Used version numbers for dependencies -->
>         <liferay.portal-service.version>6.2.5</liferay.portal-service.version>
> ```

(ideally the version change would be a separate patche, but not a blocker)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/7959)*
<!-- Sent from Reviewable.io -->

hesara commented:
OK even though would be nicer as separate patches

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/7959#-:-KYnFK6x5mEziF6CiEU1:b-iq6qhk)*
<!-- Sent from Reviewable.io -->

Commit:
Update liferay dependency to 6.2.5

Official Liferay support for FW 8 is 6.2.

Commit:
Reduce javadoc errors by adding missing dependencies

vaadin-all currently has +100 errors/warning when creating javadocs,
this reduces most of them related to missing resources to link to.

pleku commented:
Split into two patches, must merge without squashing

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/7959#-:-KYnHHZD9SCaJBFp7wnq:b-rql7pv)*
<!-- Sent from Reviewable.io -->
